Makefile

```diff
@@ -245,8 +248,15 @@ lint: buildcache
 
 PILOT_GO_BINS:=${ISTIO_OUT}/pilot-discovery ${ISTIO_OUT}/pilot-agent \
                ${ISTIO_OUT}/istioctl ${ISTIO_OUT}/sidecar-injector
-$(PILOT_GO_BINS): depend
-	bin/gobuild.sh $@ istio.io/istio/pkg/version ./pilot/cmd/$(@F)
+PILOT_GO_BINS_SHORT:=pilot-discovery pilot-agent istioctl sidecar-injector
+define pilotbuild
+$(1):
+	bin/gobuild.sh ${ISTIO_OUT}/$(1) istio.io/istio/pkg/version ./pilot/cmd/$(1)
+
+${ISTIO_OUT}/$(1):
+	bin/gobuild.sh ${ISTIO_OUT}/$(1) istio.io/istio/pkg/version ./pilot/cmd/$(1)
+endef
+$(foreach ITEM,$(PILOT_GO_BINS_SHORT),$(eval $(call pilotbuild,$(ITEM))))
```

Comment:
To be fair to @mattdelco , this is more makefile black magic. I can see that its calling functions. May be some comment would reduce the chances of someone coming in and messing up the logic.

 But I dont see the need for another pilot-go-bins-short when we already have a pilot-go-bins..

Reply:
I unrolled some of the black magic (for the other targets). 
Yes, I can use some substring function to convert go-bins to go-bins-short - but I'm not 
very good writing code in make, and I suspect others will find it hard to read too.

I'll eventually unroll this into explicit targets as well, but in a separate PR, that's not the main 
purpose of this.